Feature/9217 top traffic source driving leads

zutigrm opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #9217 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

zutigrm commented:
Note to the reviewer: The E2E tests are failing for reasons not related to this PR, but they originate from different issue that was merged, and causes E2E failures in all PR's

tofumatt commented on tests/backstop/reference/google-site-kit_KeyMetrics_TopTrafficSourceDrivingLeads_InsufficientPermissions_0_document_2_large.png:
This isn't an error state specific to this widget, so let's remove this test. We already have a _lot_ of VRTs and the size is often causing them to fail/timeout, plus it takes longer to run for every test we add. When they add value that's worth it, but this one doesn't add much, so let's remove it 🙂

tofumatt commented on tests/backstop/reference/google-site-kit_KeyMetrics_TopTrafficSourceDrivingLeads_Error_0_document_2_large.png:
This isn't an error state specific to this widget, so let's remove this test. We already have a _lot_ of VRTs and the size is often causing them to fail/timeout, plus it takes longer to run for every test we add. When they add value that's worth it, but this one doesn't add much, so let's remove it 🙂

tofumatt commented on tests/backstop/reference/google-site-kit_KeyMetrics_TopTrafficSourceDrivingLeadsWidget_ZeroData_0_document_1_medium.png:
Not sure this one is worth a VRT either, this can probably be removed 🙂

tofumatt commented on tests/backstop/reference/google-site-kit_KeyMetrics_TopTrafficSourceDrivingLeadsWidget_Ready_0_document_2_large.png:
Even this is covered by other VRTs/the base KMW VRTs, so I'd say this is _probably_ safe to remove as well.

I think our amount of VRTs is probably overkill 😅

zutigrm replied:
I kept this one only, as it still handy to have it, so we can see if something goes wrong with it, as although it displays data similarly to the base traffic metric, it still has different options that are needed in order to work.

tofumatt reviewed (changes requested):
Looks good, but I think there are some code improvements to make (mostly formatting/readability) and some VRTs to remove.

zutigrm commented on assets/js/modules/analytics-4/components/widgets/TopTrafficSourceDrivingLeadsWidget.js:
```diff
@@ -0,0 +1,277 @@
+					[ trafficSourceReportOptions ]
+			  )
+			: undefined
+	);
```
Good point, updated

zutigrm commented:
Thanks @tofumatt I updated the PR, although I removed `.scenario` for most stories, they seems not be detected as removed in my local VRT runs, so they are still present in the PR files. Do you have an idea, why that might be?

tofumatt commented:
Usually you need to run `npm run test:visualapprove` to update the VRTs, I'll try it out locally.
